python/ray/tests/test_cancel.py

```diff
@@ -125,12 +125,26 @@ def test_defer_sigint():
 
 
 def test_defer_sigint_monkey_patch():
-    # Tests that setting a SIGINT signal handler within a DeferSigint context is not
-    # allowed.
+    # Tests that setting a SIGINT signal handler within a DeferSigint
+    # context will defer setting it until the context exits.
```

Comment:
This comment seems to mean the signal handler is set after the context exit, without mentioning the signal masking. What about:

"""
Tests that the SIGINT signal handlers set within a DeferSigint is triggered at most once and only at context exit.
"""
"""

Reply:
Also, can we add a test that only the last set up of the handler is triggered? Some thing like:

```
with DeferSigint():
    signal.signal(signal.SIGINT, handler_1)
    send_self_a_signal()
    send_self_a_signal()
    signal.signal(signal.SIGINT, handler_2)
    send_self_a_signal()
    send_self_a_signal()
    assert_received_signal_times(handler_1, 0)
    assert_received_signal_times(handler_2, 0)
assert_received_signal_times(handler_1, 0)
assert_received_signal_times(handler_1, 1)
```

Reply:
done